nemo/collections/asr/data/text_to_text.py

```diff
@@ -0,0 +1,462 @@
+
+    def __init__(
+        self,
+        manifest_filepath: Union[AnyPath, List[AnyPath]],
+        speakers_filepath: Union[AnyPath, List[AnyPath]],
+        asr_tokenizer: TokenizerSpec,
+        asr_use_start_end_token: bool,
+        tts_parser: Callable,
+        tts_text_pad_id: int,
+        tts_text_normalizer: Normalizer,
+        tts_text_normalizer_call_kwargs: Dict,
+        min_words: int = 1,
+        max_words: int = 1_000_000,
+        tokenizer_workers: int = 1,
+        num_parts: int = 1,
+        current_part_index: int = 0,
+    ):
+        super().__init__()
+        # ASR tokenizer setup
+        if asr_use_start_end_token and hasattr(asr_tokenizer, 'bos_token'):
+            self.asr_bos_id = asr_tokenizer.bos_id
+
+        if asr_use_start_end_token and hasattr(asr_tokenizer, 'eos_token'):
+            self.asr_eos_id = asr_tokenizer.eos_id
+
+        if hasattr(asr_tokenizer, 'pad_token'):
+            self.asr_pad_id = asr_tokenizer.pad_id
+        else:
+            self.asr_pad_id = 0
+
+        self.asr_tokenizer = asr_tokenizer
+
+        # TTS tokenizer setup
+        self.tts_parser = tts_parser
+        self.tts_normalizer = tts_text_normalizer
+        self.tts_normalizer_kwargs = tts_text_normalizer_call_kwargs
+        self.tts_text_pad_id = tts_text_pad_id
+
+        # Load speakers
+        if isinstance(speakers_filepath, str):
+            speakers_filepath = speakers_filepath.split(",")
+        elif isinstance(speakers_filepath, Path):
+            speakers_filepath = [speakers_filepath]
+        speakers: Set[int] = set()
+        for filepath in speakers_filepath:
+            with open(Path(filepath).expanduser(), "r") as f:
+                speakers.update(map(int, f.read().split()))
+        self.speakers = np.asarray(sorted(speakers))
+        logging.info(f"Loaded {len(self.speakers)} speakers")
+
+        # Load manifest
+        if isinstance(manifest_filepath, str):
+            manifest_filepath = manifest_filepath.split(",")
+        elif isinstance(manifest_filepath, Path):
+            manifest_filepath = [manifest_filepath]
+        self.manifest_paths = [Path(filepath) for filepath in manifest_filepath]
+
+        num_skipped_words = 0
+        num_skipped_utterances = 0
+        asr_texts = []
+        tts_texts = []
+        need_normalization = False
+
+        for manifest_path in self.manifest_paths:
+            for tmp_item in tqdm(_iterate_manifest(manifest_path)):
+                text = tmp_item["text"]
+                num_words = len(text.split())
+                # skip if number of works not in desired range
+                # TODO: maybe it would be valuable to sample sub-utterances from long utterances
+                if not (min_words <= num_words <= max_words):
+                    num_skipped_words += num_words
+                    num_skipped_utterances += 1
+                    continue
+                asr_texts.append(tmp_item["text"])
+                try:
```

Comment:
`if "tts_text_normalized" in tmp_item`?
Should be faster than try/except.

Reply:
Changed